Error while running sync:  cannot sync cluster config

During a cluster deploy I saw the following error:

```
2018/07/27 13:54:10 Error while syncing: cannot sync cluster config: serviceaccounts "default-rolebindings-controller" already exists
```

```
+ go run cmd/sync/sync.go -run-once=true
2018/07/27 13:50:31 Sync process started!
2018/07/27 13:54:06 Update Namespace/default
2018/07/27 13:54:06 - Object.map[spec].map[finalizers].slice[1]: <no value> != openshift.io/origin
2018/07/27 13:54:06 - Object.map[metadata].map[annotations]: <does not have key> != map[openshift.io/node-selector:]
2018/07/27 13:54:06 Skip Namespace/kube-public
2018/07/27 13:54:06 Create Namespace/kube-service-catalog
2018/07/27 13:54:06 Skip Namespace/kube-system
2018/07/27 13:54:06 Update Namespace/openshift
2018/07/27 13:54:06 - Object.map[spec].map[finalizers].slice[1]: <no value> != openshift.io/origin
2018/07/27 13:54:06 Create Namespace/openshift-ansible-service-broker
2018/07/27 13:54:06 Create Namespace/openshift-azure
2018/07/27 13:54:06 Skip Namespace/openshift-infra
2018/07/27 13:54:06 Create Namespace/openshift-metrics
2018/07/27 13:54:06 Update Namespace/openshift-node
2018/07/27 13:54:06 - Object.map[metadata].map[annotations]: <does not have key> != map[openshift.io/node-selector:]
2018/07/27 13:54:07 Create Namespace/openshift-sdn
2018/07/27 13:54:07 Create Namespace/openshift-template-service-broker
2018/07/27 13:54:07 Create Namespace/openshift-web-console
2018/07/27 13:54:07 Create ServiceAccount/default/default
2018/07/27 13:54:07 Create ServiceAccount/default/registry
2018/07/27 13:54:07 Create ServiceAccount/default/router
2018/07/27 13:54:07 Create ServiceAccount/kube-service-catalog/service-catalog-apiserver
2018/07/27 13:54:07 Create ServiceAccount/kube-service-catalog/service-catalog-controller
2018/07/27 13:54:08 Create ServiceAccount/kube-system/attachdetach-controller
2018/07/27 13:54:08 Skip ServiceAccount/kube-system/certificate-controller
2018/07/27 13:54:08 Skip ServiceAccount/kube-system/clusterrole-aggregation-controller
2018/07/27 13:54:08 Skip ServiceAccount/kube-system/cronjob-controller
2018/07/27 13:54:08 Create ServiceAccount/kube-system/daemon-set-controller
2018/07/27 13:54:08 Create ServiceAccount/kube-system/deployment-controller
2018/07/27 13:54:08 Skip ServiceAccount/kube-system/disruption-controller
2018/07/27 13:54:08 Create ServiceAccount/kube-system/endpoint-controller
2018/07/27 13:54:08 Skip ServiceAccount/kube-system/generic-garbage-collector
2018/07/27 13:54:08 Skip ServiceAccount/kube-system/job-controller
2018/07/27 13:54:09 Create ServiceAccount/kube-system/namespace-controller
2018/07/27 13:54:09 Skip ServiceAccount/kube-system/node-controller
2018/07/27 13:54:09 Create ServiceAccount/kube-system/persistent-volume-binder
2018/07/27 13:54:09 Skip ServiceAccount/kube-system/pod-garbage-collector
2018/07/27 13:54:09 Skip ServiceAccount/kube-system/pv-protection-controller
2018/07/27 13:54:09 Skip ServiceAccount/kube-system/pvc-protection-controller
2018/07/27 13:54:09 Skip ServiceAccount/kube-system/replicaset-controller
2018/07/27 13:54:09 Skip ServiceAccount/kube-system/replication-controller
2018/07/27 13:54:09 Skip ServiceAccount/kube-system/resourcequota-controller
2018/07/27 13:54:09 Skip ServiceAccount/kube-system/service-account-controller
2018/07/27 13:54:09 Skip ServiceAccount/kube-system/service-controller
2018/07/27 13:54:09 Skip ServiceAccount/kube-system/statefulset-controller
2018/07/27 13:54:09 Create ServiceAccount/openshift-ansible-service-broker/asb
2018/07/27 13:54:09 Create ServiceAccount/openshift-ansible-service-broker/asb-client
2018/07/27 13:54:09 Create ServiceAccount/openshift-infra/bootstrap-autoapprover
2018/07/27 13:54:09 Skip ServiceAccount/openshift-infra/build-config-change-controller
2018/07/27 13:54:10 Skip ServiceAccount/openshift-infra/build-controller
2018/07/27 13:54:10 Skip ServiceAccount/openshift-infra/cluster-quota-reconciliation-controller
2018/07/27 13:54:10 Create ServiceAccount/openshift-infra/default-rolebindings-controller
2018/07/27 13:54:10 Error while syncing: cannot sync cluster config: serviceaccounts "default-rolebindings-controller" already exists
+ az group deployment wait -g kwoodsontest -n azuredeploy --created --interval 10
+ KUBECONFIG=_data/_out/admin.kubeconfig
+ go run cmd/healthcheck/healthcheck.go
panic: unexpected error code 403 from console

goroutine 1 [running]:
main.main()
	/home/kwoodson/go/src/github.com/openshift/openshift-azure/cmd/healthcheck/healthcheck.go:53 +0x44
exit status 2
```

--- a/pkg/addons/client.go
+++ b/pkg/addons/client.go
@@ -13,6 +13,7 @@ import (
 	"k8s.io/apimachinery/pkg/api/meta"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/client-go/discovery"
 	"k8s.io/client-go/dynamic"
 	"k8s.io/client-go/rest"
@@ -151,6 +152,15 @@ func write(dyn dynamic.ClientPool, grs []*discovery.APIGroupResources, o *unstru
 		if kerrors.IsNotFound(err) {
 			log.Println("Create " + KeyFunc(o.GroupVersionKind().GroupKind(), o.GetNamespace(), o.GetName()))
 			_, err = dc.Resource(res, o.GetNamespace()).Create(o)
+			if kerrors.IsAlreadyExists(err) {
+				// The "hot path" in write() is Get, check, then maybe Update.
+				// Optimising for this has the disadvantage that at cluster
+				// creation we can race with API server or controller
+				// initialisation. Between Get returning NotFound and us trying
+				// to Create, the object might be created.  In this case we
+				// return a synthetic Conflict to force a retry.
+				err = kerrors.NewConflict(schema.GroupResource{Group: res.Group, Resource: res.Name}, o.GetName(), errors.New("synthetic"))
+			}
 			return
 		}
 		if err != nil {

